### DISCUSSION TITLE
Nested resources menu: how to get a link to the parent resource list?

### DISCUSSION THREAD

**Discussion by Bastes** [activeadmin/activeadmin#7877] [Help]
This is a question (that could get to be a feature request) about menu generation for nested resources (using `belongs_to`).

So, what I have is something along those lines:
```ruby
ActiveAdmin.register ParentResource do
  ...
end

ActiveAdmin.register ChildResource do
  belongs_to :parent_resource
  navigation_menu :parent_resource
  ...
end
```
The menu I get on the `ParentResource` index page:
```
[ParentResources]
```
(so far so good)
On the `ChildResource` index though I would like:
```
[ParentResources|ChildResources]
```
Though what I get is just:
```
[ChildResources]
```

I tried adding:
```ruby
ActiveAdmin.register ParentResource do
  navigation_menu :parent_resource
end
```
But that bugs on the ```ChildResource``` index page because it tries to generate the ```ChildResource``` link without a ```params[:parent_resource_id]``` and on the ```ChildResource``` page it somehow highlights the ```ParentResources``` link instead of the ```ChildResource``` link.

I'm not as intimate with the ActiveAdmin codebase as to know for sure there isn't a way to do what I'd like to do but the documentation does not seem to indicate any way to do this and the researches I've made so far didn't hold fruit. So, is there already a way to achieve this behaviour I havent found or is it something that is not possible so far? And if not, is it a reasonable use case to consider? I'd make a PR if I knew where to look to achieve this :sweat_smile:

**Comment by benoittgt** [comment]
One option is to set the `if` option in the menu.

For example:
```ruby
ActiveAdmin.register ParentResource do
  menu :false
end

ActiveAdmin.register ChildResource do
  belongs_to :parent_resource
  navigation_menu :parent_resource
  menu(
    label: I18n.t(:menu, scope: [[:ChildResource, :action]]),
    priority: 2,
    if: proc { params[:my_param_requested_for_url_generation].present? },
  )
end
```

This is hacky, but it works. By default ActiveAdmin should disable menu if the parent is in `menu :false`.
